packages/tabs/src/index.js

```diff
@@ -1,9 +1,35 @@
-import React, { cloneElement, useState, useRef, forwardRef } from "react";
+import React, {
+  useState,
+  useRef,
+  useContext,
+  forwardRef,
+  createContext
+} from "react";
 import { node, func, number } from "prop-types";
 import warning from "warning";
 import { wrapEvent, useUpdateEffect, makeId, useForkedRef } from "@reach/utils";
 import { useId } from "@reach/auto-id";
 
+const createNamedContext = name => {
+  const context = createContext();
+  context.displayName = name;
+
+  return context;
+};
+
+const TabsContext = createNamedContext("Tabs");
+const TabContext = createNamedContext("Tab");
+const TabPanelContext = createNamedContext("TabPanel");
+
+export const useTabState = () => {
+  const index = useContext(TabContext);
+  const { selectedIndex } = useContext(TabsContext);
+
+  return {
+    isSelected: index === selectedIndex
+  };
+};
```

Comment:
Don't know if Reach expects clients to be on React 16.8+, but we could add in a development warning for it the way [React Router](https://github.com/ReactTraining/react-router/blob/ea44618e68f6a112e48404b2ea0da3e207daf4f0/packages/react-router/modules/hooks.js#L10-L15) does if we'd like

```js
export const useTabState = () => {
  if (__DEV__) {
    invariant(
      typeof useContext === "function",
      "You must use React >= 16.8 in order to use useTabState()"
    );
  }

  const index = useContext(TabContext);
  const { selectedIndex } = useContext(TabsContext);

  return {
    isSelected: index === selectedIndex
  };
};
```

Reply:
Definitely a brain fart moment here … obviously we require 16.8 for this repo, most of the components are using hooks.